habitat-lab/habitat/tasks/rearrange/utils.py

```diff
@@ -516,11 +508,10 @@ def get_robot_spawns(
             if did_collide:
                 break
 
+        sim.set_state(state)
```

Comment:
Is there a cost to calling `sim.set_state(state)` ? If that is the case, sim.set_state should be called only before returning and not inside this for loop. I think this way it will be called every iteration instead of just once.

Reply:
This was used to reset the robot position before re-sampling I think. Probably not necessary.